File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.py
# 어떤 번호가 다른 번호의 접두인 경우가 있으면 false, 아니면 true를 반환
def solution(phone_book):
    hash_map = {}
    for phone_nums in phone_book:
        hash_map[phone_nums] = 1
    
    for nums in phone_book:
        arr = ''
        for num in nums:
            arr += num
            
            if arr in hash_map and arr != nums:
                return False
    return True

File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.js
function solution(phone_book) {
    const hash_map = {}
    
    for (let phone of phone_book) {
        hash_map[phone] = 1
    }
    
    for (let phone of phone_book) {
        let arr = '';
        for (let num of phone) {
            arr += num;
            
            if (arr in hash_map && arr !== phone) {
                return false
            }
        }
    }
    return true 
}